playbooks/common/openshift-cluster/upgrades/v3_0_to_v3_1/upgrade.yml

```diff
@@ -74,6 +74,11 @@
 
   - set_fact:
       g_new_version: "{{ g_aos_versions.curr_version.split('-', 1).0 if g_aos_versions.avail_version is none else g_aos_versions.avail_version.split('-', 1).0 }}"
+    when: openshift_pkg_version is not defined
+
+  - set_fact:
+      g_new_version: "{{ openshift_pkg_version | replace('-','') }}"
+    when: openshift_pkg_version is defined
```

Comment:
This is just another bug fix.  Previously if you set openshift_pkg_version to something less than the latest the correct RPM would be installed but the router would be updated to the latest image.